### DISCUSSION TITLE
同步时是否应该向 rpc 订阅者广播 block 消息？

### DISCUSSION THREAD

**Discussion by jolestar** [starcoinorg/starcoin#2378] [Technical]
当前的策略是正在同步时，不对 rpc 的订阅者广播新 block 消息，直到同步完广播最后一个区块。

发现的问题：如果用在数据同步场景，比如区块浏览器，会发现如果节点处于同步状态时，区块浏览器的数据一直不更新。

**Comment by GXKe** [comment]
同步状态中可以广播，在正常挖矿过程中一般链是同步的，不会落后导致进入同步状态场景

**Reply by jolestar** [reply]
> 同步状态中可以广播，在正常挖矿过程中一般链是同步的，不会落后导致进入同步状态场景

同步中的区块会触发无效的挖矿任务，这个也没关系吗？

**Reply by GXKe** [reply]
没有关系，这属于异常状态。
同步后新任务会覆盖掉无效的
